docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+
+![ArchitectureSequence.png](images%2FArchitectureSequence.png)
+
+### Modules component
+
+### Data component
+
+{insert diagrams}
+
+- stores the financial report data, such as financial statement objects
+
+- Monthly expenditures are stored in the form of a Hash Table, mapping the Year and Month to the expenditures
+
+#### Design considerations
+
+- To prevent slow processing of the program when there are many entries, a Hash Table is used to keep track of the
+  expenditures for the month.
+  - We believe that the "Set Budget" feature will be commonly used as it is 1 of the key aspects of a Financial Tracker. 
+    Hence, it would be feasible to implement an additional storage 
+    such that common addition operations and retrieval of information can be done in Amortised O(1).
+  - The Key used is the number of months from the Year 0000, or more precisely calculated by (Year * 12 + Month). 
+    This is to allow for fast processing and collision-free information collection.
+
+
+### Command component
+
+{insert diagrams}
+
+1. When a command is parsed, a command object specific to the command given will be created, with the necessary
+   parameters stored as attributes in the command specific class
+2. RainyDay will then call method execute() in the command class to return a CommandResult object, which contains
+   the output to be shown to user
+
+### Exceptions component
+
+## Implementation
+
+### Maintaining of Financial Report
+
+- `financialReport` is achieved with the use of an ArrayList collection of the object `FinancialStatement` to model a
+  financial report containing a list of financial statements.
+- A 'FinancialStatement` object contains the following attributes:
+    - `description`: represents the description of a transaction, stored as a string
+    - `flowDirection`: represents the direction of a transaction, stored as an enum of `FlowDirection`: `INFLOW` or
+      `OUTFLOW`
+    - `value`: represents the amount tagged with the transaction, stored as a double
+    - `category`: represents the type of transaction, stored as a string
+    - `date`: represents the date to be tagged of transaction, stored as LocalDate object
+
+### Adding an entry `add`
+
+- When a command is given to add a statement, the command is first parsed to check whether it follows the format of an
+  add command: `add -DIRECTION DESCRIPTION $VALUE -c CATEGORY -date DD/MM/YYYY` with the use of regex pattern
+    - Details on implementation for parsing and command fields are documented below
+- Commands in the correct format will then be parsed to extract the relevant information, and an `AddCommand` object
+  will be created with the relevant attributes
+- `RainyDay` will then call `execute` method in `Command`, where the transaction will be added into the financial report
+
+The sequence diagram for the implementation of add is as shown below:
+
+![AddCommandSequenceDiagram.png](images\DeveloperGuide\AddCommandSequenceDiagram.png)
```

Comment:
Would it be easier to understand if the return result to the user is provided in explanation?